src/RuleMatcher.h

```diff
@@ -168,6 +169,35 @@ class RuleEndpointState {
     RuleEndpointState(analyzer::Analyzer* arg_analyzer, bool arg_is_orig, RuleEndpointState* arg_opposite,
                       analyzer::pia::PIA* arg_PIA);
 
+    // Tracking pattern matches for a given Rule.
+    class RulePatternMatch {
```

Comment:
Optional: if all members are public, I personally like making it a `struct` to indicate so.